### PR TITLE
Moved conditions from configuration controller to report controller

### PR DESCRIPTION
**Pull request by HlinaCZ**
Resolves: #139

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

**Comment by phoracek** [comment]
ci test please

**Comment by phoracek** [comment]
ci test please

**Comment by phoracek** [comment]
ci add to whitelist

**Comment by phoracek** [comment]
ci test please

File: pkg/controller/nodenetworkstateconfiguration/nodenetworkstateconfiguration_controller.go [hunk 1]
```diff
@@ -200,21 +139,9 @@ func (r *ReconcileNodeNetworkStateConfiguration) Reconcile(request reconcile.Req
 
 	nmstateOutput, err := nmstate.ApplyDesiredState(instance)
 	if err != nil {
-		errmsg := fmt.Errorf("error reconciling nodenetworkstate at desired state apply: %s, %v", nmstateOutput, err)
```
Comment (qinqon) [inline pkg/controller/nodenetworkstateconfiguration/nodenetworkstateconfiguration_controller.go]: Don't remove it from here, let's have it at both controllers, it's useful for debugging purposes and to really wait for rollback to finish.

File: test/e2e/rollback_test.go [hunk 2]
```diff
@@ -58,9 +50,8 @@ var _ = Describe("rollback", func() {
 		It("should rollback failed state configuration", func() {
 			updateDesiredState(linuxBrUpNoPorts(bridge1))
 			for _, node := range nodes {
-				By("Wait for reconcile to fail")
```
Comment (qinqon) [inline test/e2e/rollback_test.go]: ditto

File: test/e2e/rollback_test.go [hunk 3]
```diff
@@ -88,9 +79,8 @@ var _ = Describe("rollback", func() {
 		})
 		It("should rollback to a good gw configuration", func() {
 			for _, node := range nodes {
-				By("Wait for reconcile to fail")
```
Comment (qinqon) [inline test/e2e/rollback_test.go]: ditto

File: test/e2e/rollback_test.go [hunk 4]
```diff
@@ -106,20 +96,3 @@ var _ = Describe("rollback", func() {
 		})
 	})
 })
-
-func nodeNetworkStateConditionStatus(node string, expectedConditionType nmstatev1alpha1.NodeNetworkStateConditionType) corev1.ConditionStatus {
```
Comment (qinqon) [inline test/e2e/rollback_test.go]: ditto

**Comment by phoracek** [comment]
@qinqon  @HlinaCZ let's work on https://github.com/nmstate/kubernetes-nmstate/pull/236 and merge this one only once we have a proper replacement for configuration status reporting.

**Comment by phoracek** [comment]
Recovered tests:

```go
package e2e

import (
	. "github.com/onsi/ginkgo"
	. "github.com/onsi/gomega"

	corev1 "k8s.io/api/core/v1"

	nmstatev1alpha1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1alpha1"
)

var _ = Describe("NodeNetworkStateCondition", func() {
	Context("when cluster is initialized", func() {
		BeforeEach(func() {
			deleteNodeNeworkStates()
		})
		It("should have Available ConditionType set to true", func() {
			for _, node := range nodes {
				checkCondition(node, nmstatev1alpha1.NodeNetworkStateConditionAvailable).Should(
					Equal(corev1.ConditionTrue),
				)
				checkCondition(node, nmstatev1alpha1.NodeNetworkStateConditionFailing).Should(
					Equal(corev1.ConditionFalse),
				)
			}
		})
	})

	Context("when nmstatectl show fails", func() {
		BeforeEach(func() {
			By("Rename nmstatectl to nmstatectl.bak to force failure during nmstatectl show")
			runAtPods("mv", "/usr/bin/nmstatectl", "/usr/bin/nmstatectl.bak")
			deleteNodeNeworkStates()
		})
		AfterEach(func() {
			By("Rename nmstatectl.bak to nmstatectl to have functional nmstatectl")
			runAtPods("mv", "/usr/bin/nmstatectl.bak", "/usr/bin/nmstatectl")
		})
		It("should fail", func() {
			for _, node := range nodes {
				checkCondition(node, nmstatev1alpha1.NodeNetworkStateConditionFailing).Should(
					Equal(corev1.ConditionTrue),
				)
				checkCondition(node, nmstatev1alpha1.NodeNetworkStateConditionAvailable).Should(
					Equal(corev1.ConditionFalse),
				)
			}
		})
	})
})
```

**Comment by phoracek** [comment]
Replaced by https://github.com/nmstate/kubernetes-nmstate/pull/323

**Pull request closed without merging**